### PR TITLE
Filter environment markers in dependency mismatch warnings

### LINKED ISSUE #21765: [BUG] `warn_dependency_requirement_mismatches` emits spurious warnings for requirements mismatches over model logged with uv detection
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Issues Policy acknowledgement

- [x] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### MLflow version

- Client: MLflow master

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MacOS
- **Python version**: 3.10

### Describe the problem

`warn_dependency_requirement_mismatches` emits spurious warnings for requirements mismatches over model logged with uv detection:

For model logged with uv detection, the requirements.txt file (generated by `uv export`) might contain environment marker like:

```
numpy==2.2.6; python_full_version < "3.11"
numpy==2.4.3; python_full_version >= "3.11"
nvidia-nccl-cu12==2.29.7; sys_platform == "linux"
```

When running model inference with uv env-manager, `warn_dependency_requirement_mismatches` does not handle this case well and raise spurious warnings like:

```
 - numpy (current: 2.2.6, required: numpy==2.4.3; python_full_version >= "3.11")
 - nvidia-nccl-cu12 (current: uninstalled, required: nvidia-nccl-cu12==2.29.7; sys_platform == "linux")
```

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

Environment: my MacOS with **python 3.10**

1. checkout mlflow master,
2. then run `pip install -e .` ,
3. then enter `examples/uv-dependency-management`,
4. then run `python log_model.py`

5. create a python script at `/tmp/p1.py` as follows:

```
import mlflow
from mlflow.models import predict
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split

iris = load_iris()
X_train, X_test, y_train, y_test = train_test_split(
    iris.data, iris.target, test_size=0.2, random_state=42
)
model_uri = <model uri to the model logged by example_groups_and_extras>
predict(
    model_uri=model_uri,
    input_data=X_test,
    env_manager="uv",
)
```
run `python /tmp/p1`

and it emits warning like
```
2026/03/17 22:10:44 WARNING mlflow.utils.requirements_utils: Detected one or more mismatches between the model's dependencies and the current Python environment:
 - colorama (current: uninstalled, required: colorama==0.4.6; sys_platform == "win32")
 - contourpy (current: 1.3.2, required: contourpy==1.3.3; python_full_version >= "3.11")
 - greenlet (current: uninstalled, required: greenlet==3.3.2; platform_machine == "AMD64" or platform_machine == "WIN32" or platform_machine == "aarch64" or platform_machine == "amd64" or platform_machine == "ppc64le" or platform_machine == "win32" or platform_machine == "x86_64")
 - numpy (current: 2.2.6, required: numpy==2.4.3; python_full_version >= "3.11")
 - nvidia-nccl-cu12 (current: uninstalled, required: nvidia-nccl-cu12==2.29.7; sys_platform == "linux")
 - pywin32 (current: uninstalled, required: pywin32==311; sys_platform == "win32")
 - scikit-learn (current: 1.7.2, required: scikit-learn==1.8.0; python_full_version >= "3.11")
 - scipy (current: 1.15.3, required: scipy==1.17.1; python_full_version >= "3.11")
 - waitress (current: uninstalled, required: waitress==3.0.2; sys_platform == "win32")
 - xgboost (current: uninstalled, required: xgboost==3.2.0)
```

### Stack trace

```
REPLACE_ME
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow model serving, deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/prompt`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/uiux`: Front-end, user experience, plotting
- [ ] `area/docs`: MLflow documentation pages

### PR DESCRIPTION
**Pull request by debu-sinha**
### Related Issues/PRs

Closes #21765

### What changes are proposed in this pull request?

`warn_dependency_requirement_mismatches` emits spurious warnings for requirements with environment markers from uv export. For example:

```
numpy==2.4.3; python_full_version >= "3.11"
nvidia-nccl-cu12==2.29.7; sys_platform == "linux"
```

These should be silently skipped when the markers don't match the current environment. The existing check in `_check_requirement_satisfied` (line 624) handles individual requirements, but this adds a defensive pre-filter at the caller level to skip non-applicable requirements early before version checking.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

Three new tests:
1. `test_warn_dependency_skips_non_matching_environment_markers` - no warning for non-matching markers
2. `test_warn_dependency_fires_for_matching_marker_with_missing_package` - warning fires when marker matches but package missing
3. `test_warn_dependency_skips_uv_style_dual_markers` - uv dual-marker format only warns for matching marker

Manual test: called `warn_dependency_requirement_mismatches` with exact requirement strings from the issue on Python 3.10/macOS. Non-matching markers correctly suppressed.

### Does this PR require documentation update?

- [x] No.

### Does this PR require updating the [MLflow Skills](https://github.com/mlflow/skills) repository?

- [x] No.

#### Is this a user-facing change?

- [x] Yes. Fix spurious dependency mismatch warnings when loading models logged with uv environment markers.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes

#### Is this PR a critical bugfix or security fix that should go into the next patch release?

- [ ] This PR is critical and needs to be in the next patch release
- [x] This PR can wait for the next minor release

### CODE REVIEW

**Comment by debu-sinha** [comment]
Closing PR until the issue can be reproduced.
